### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException: 1

### ISSUE THREAD

**Issue by baev** [allure-framework/allure-cucumberjvm#17]
```
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.testSuiteStarted(AllureRunListener.java:185)
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.getSuiteUid(AllureRunListener.java:316)
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.testStarted(AllureRunListener.java:261)
```

When run tests (adaptor version 1.3)

The problem in the following line:

``` java
Stories story = getStoriesAnnotation(new String[]{annotationParams[1].split(":")[1].trim()});
```

**Comment by clicman** [comment]
@tiffanyplourde could you provide the feature on which failure can be reproduced? could you provide the feature on which failure can be reproduced?

**Comment by Jiwari** [comment]
I'm having a similar error when I execute a cucumber dry run:
`test -Dcucumber.options="-d"`

Not sure if related to this same issue, if not I'll open a new issue.

Stacktrace:
`Tests run: 75, Failures: 0, Errors: 6, Skipped: 60, Time elapsed: 0.263 sec <<< FAILURE! - in feature.ppf2.contratarccr.ContratarCCRTest
Test mechanism  Time elapsed: 0.007 sec  <<< ERROR!
java.lang.ArrayIndexOutOfBoundsException: 1
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.testSuiteStarted(AllureRunListener.java:183)
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.getSuiteUid(AllureRunListener.java:321)
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.startFakeTestCase(AllureRunListener.java:336)
    at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.testIgnored(AllureRunListener.java:287)
    at org.junit.runner.notification.SynchronizedRunListener.testIgnored(SynchronizedRunListener.java:77)
    at org.junit.runner.notification.RunNotifier$6.notifyListener(RunNotifier.java:174)
    at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:72)
    at org.junit.runner.notification.RunNotifier.fireTestIgnored(RunNotifier.java:171)
    at org.junit.internal.runners.model.EachTestNotifier.fireTestIgnored(EachTestNotifier.java:46)
    at cucumber.runtime.junit.JUnitReporter.result(JUnitReporter.java:89)
    at cucumber.runtime.Runtime.runStep(Runtime.java:293)
    at cucumber.runtime.model.StepContainer.runStep(StepContainer.java:44)
    at cucumber.runtime.model.StepContainer.runSteps(StepContainer.java:39)
    at cucumber.runtime.model.CucumberScenario.run(CucumberScenario.java:44)
    at cucumber.runtime.junit.ExecutionUnitRunner.run(ExecutionUnitRunner.java:91)
    at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:63)
    at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:18)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    at cucumber.runtime.junit.FeatureRunner.run(FeatureRunner.java:70)
    at cucumber.api.junit.Cucumber.runChild(Cucumber.java:93)
    at cucumber.api.junit.Cucumber.runChild(Cucumber.java:37)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    at cucumber.api.junit.Cucumber.run(Cucumber.java:98)
    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
    at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)`

**Comment by clicman** [comment]
@Jiwari cause of original error is unknown because reporter didn not provide additional info.
But in case of dry run allure adaptor shuld be turned off i think, because it is for debugging purpose. Am I right?

**Comment by gvvinblade** [comment]
Hi guys, Looks like the bug isn't in the Allure adapter, but in the METHOD_AND_CLASS_NAME_PATTERN regexp in org.junit.runner.Description. It can't work fine with features files where round brackets are present in a scenario name. Next piece of changes resolved the issue, at least for me:

```
 public class AllureRunListener extends RunListener {

     //------------other stuff----------------

       private String extractClassName(Description description) {
       String displayName = description.getDisplayName();
       Pattern pattern = Pattern.compile("\\(\\|(.*)\\|\\)");
       Matcher matcher = pattern.matcher(displayName);
       if (matcher.find()) {
           return "|" + matcher.group(1) + "|";
       }
       return getClassName(description); //instead of return description.getClassName();
    }

    //new method
    private String getClassName(Description description) {
       Pattern pattern = Pattern.compile("[^\\(]*(\\((.*)\\))[^\\)]*");
       Matcher matcher = pattern.matcher(description.getDisplayName());
       if(matcher.find()){
           return matcher.group(2);
       }
       return description.getClassName();
    }

    //------------other stuff---------------

  }
```

 `

**Comment by clicman** [comment]
@gvvinblade yes it is known limitation and I'm too lazy to fix it :) 
But thank you for remind. I will try to fix it in few days. :)

**Comment by clicman** [comment]
After some research I figured out what it leads to NPE in multithreaded execution case. See braces branch. https://github.com/allure-framework/allure-cucumber-jvm-adaptor/tree/braces

**Comment by SButterfly** [comment]
Caught the same exception by running `mvn clean test -Dcucumper.options="--name 'Some name' -t @debug"`

```
2016-03-05 12:40:25 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19:test (default-test) on project some_project: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19:test failed: There was an error in the forked process
2016-03-05 12:40:25 [ERROR] org.apache.maven.surefire.testset.TestSetFailedException: java.lang.ArrayIndexOutOfBoundsException: 1
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.common.junit4.JUnit4RunListener.rethrowAnyTestMechanismFailures(JUnit4RunListener.java:209)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:194)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:286)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:240)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
2016-03-05 12:40:25 [ERROR] Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
2016-03-05 12:40:25 [ERROR] at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.testSuiteStarted(AllureRunListener.java:185)
2016-03-05 12:40:25 [ERROR] at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.getSuiteUid(AllureRunListener.java:316)
2016-03-05 12:40:25 [ERROR] at ru.yandex.qatools.allure.cucumberjvm.AllureRunListener.testStarted(AllureRunListener.java:261)
2016-03-05 12:40:25 [ERROR] at org.junit.runner.notification.RunNotifier$3.notifyListener(RunNotifier.java:115)
2016-03-05 12:40:25 [ERROR] at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:61)
2016-03-05 12:40:25 [ERROR] at org.junit.runner.notification.RunNotifier.fireTestStarted(RunNotifier.java:112)
2016-03-05 12:40:25 [ERROR] at org.junit.internal.runners.ErrorReportingRunner.runCause(ErrorReportingRunner.java:61)
2016-03-05 12:40:25 [ERROR] at org.junit.internal.runners.ErrorReportingRunner.run(ErrorReportingRunner.java:35)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:344)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:269)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:240)
2016-03-05 12:40:25 [ERROR] at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:184)
2016-03-05 12:40:25 [ERROR] ... 3 more
2016-03-05 12:40:25 [ERROR] -> [Help 1]
2016-03-05 12:40:25 [ERROR] 
2016-03-05 12:40:25 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2016-03-05 12:40:25 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2016-03-05 12:40:25 [ERROR] 
2016-03-05 12:40:25 [ERROR] For more information about the errors and possible solutions, please read the following articles:
2016-03-05 12:40:25 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

**Comment by SButterfly** [comment]
Sorry, my case is invalid cause cucumber doesn't allow user to run simultaneously with name and tag params.

Valid command line will be: `mvn clean test -Dcucumper.options="--name 'Some name'"`

**Comment by Jiwari** [comment]
Recently I discovered that the same error is given when using `maven-surefire-plugin` v2.19.1, when there is `Scenario Outline` on the execution.

If you want to reproduce just use the [example project](https://github.com/allure-examples/allure-cucumber-jvm-example) and change the surefire version to v2.19.1.

I haven't found any solution yet, but I believe it is important to mention this issue.
As a note, the execution runs perfectly with surevire v2.18.1.

**Comment by clicman** [comment]
Yes. It's a known surefire issue. After NY holidays I plan to refactor the
adaptor, to to avoid all this issues.

24 дек. 2016 г. 7:24 PM пользователь "Eduardo Wermuth" <
notifications@github.com> написал:

**Comment by gvvinblade** [comment]
Try this one: 
https://github.com/kirlionik/allure-cucumber-plugin

It works) I've write this plugin with my frnd to solve all the issues above

**Comment by clicman** [comment]
@gvvinblade wow! it should replace original adaptor in allure.

**Comment by gvvinblade** [comment]
@clicman It's under Apache license, if you wish, you can use it as the basis for a new version of the adapter)

**Comment by clicman** [comment]
Fixed in master. Will available in 1.6 release
